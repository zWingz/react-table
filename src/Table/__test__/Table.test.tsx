import * as React from 'react'
import { shallow } from 'enzyme'
import {Table, BaseTable} from '..'
import { default as dataSource } from './DataSource'
import { ColumnProps, TableProp } from '../module'

describe('test table', () => {
  type TestDataType = ColumnProps<typeof dataSource[0]>[]
  const columns: TestDataType = [{
    title: 'id',
    dataIndex: 'id'
  }, {
    title: 'createTime',
    dataIndex: 'createTime'
  }, {
    title: <span>jsx title</span>,
    dataIndex: 'name'
  }, {
    title: 'title',
    dataIndex: 'title',
    className: 'th-custom-class'
  }]
  it('base render', () => {
    const wrapper = shallow<TableProp, any>(<Table dataSource={dataSource} columns={columns} rowKey='id'/>)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find(BaseTable)).toHaveLength(1)
    expect(wrapper.find('#fixed-table-magic')).toHaveLength(1)
    expect(wrapper.state().top).toEqual(0)
    expect(wrapper.state().paddingRight).toEqual(0)
    expect(wrapper.state().paddingLeft).toEqual(0)
    const ins = wrapper.instance() as any
    expect(ins.fixedLeft).toBeFalsy()
    expect(ins.fixedRight).toBeFalsy()
    expect(ins.cacheColumns).toEqual(columns)
  })
})
