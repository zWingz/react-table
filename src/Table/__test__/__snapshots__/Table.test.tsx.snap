// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test table base render 1`] = `
<Fragment>
  <div
    className="fixed-table-container"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <HorizontalScrollBar
      className="flex-grow"
      offsetBottom={5}
    >
      <BaseTable
        columns={
          Array [
            Object {
              "dataIndex": "id",
              "title": "id",
            },
            Object {
              "dataIndex": "createTime",
              "title": "createTime",
            },
            Object {
              "dataIndex": "name",
              "title": <span>
                jsx title
              </span>,
            },
            Object {
              "className": "th-custom-class",
              "dataIndex": "title",
              "title": "title",
            },
          ]
        }
        dataSource={
          Array [
            Object {
              "createTime": "1976-04-02",
              "id": "440000200106140115",
              "name": "尹娟",
              "title": "目派与响清",
            },
            Object {
              "createTime": "1978-12-17",
              "id": "130000199202167395",
              "name": "程丽",
              "title": "无北共外",
            },
            Object {
              "createTime": "2012-12-06",
              "id": "640000197411167747",
              "name": "何桂英",
              "title": "周切程",
            },
            Object {
              "createTime": "1990-07-18",
              "id": "440000197608270476",
              "name": "黄杰",
              "title": "一和下",
            },
          ]
        }
        getRef={
          Object {
            "current": null,
          }
        }
        rowKey="id"
        style={Object {}}
        top={0}
      />
    </HorizontalScrollBar>
    <ContextProvider
      value={Array []}
    />
  </div>
  <div
    id="fixed-table-magic"
    style={
      Object {
        "position": "fixed",
        "zIndex": 1,
      }
    }
  />
</Fragment>
`;
